bin/cli.py

```diff
@@ -11,15 +14,24 @@ def main():
     argv = sys.argv
     if len(argv) < 2 or argv[1] == "help":
         BaseCommand().help()
-        return False
+        return
 
     command = get_command_class(argv)
 
-    if command is not False:
+    if command:
         instance = command()
-        return instance.handle()
-    BaseCommand().help()
-    return False
+        try:
+            instance.sdk.login()
```

Comment:
Why put this in here? There are commands like EventsCommand that do a single auth() and pass that along to the classes that need it, and do not use a self.sdk variable at all

Reply:
Because it will raise an ApiAuthenticateError if there are no credentials. Therefore initializing a command-instance(). By doing it this way the initialization of the command-class will not fail, hence you can initialize it outside the try/catch block, and you can refer to the command-instance when you catch the exceptions. If the command instance would call self.sdk.login on init(), you would have needed an outer try/catch block for handling the ApiAuthenticateError